### PR TITLE
QBFT to send prepare when we receive a proposal

### PR DESCRIPTION
**Pull request by jframe**
Signed-off-by: Jason Frame <jasonwframe@gmail.com>

## PR description
Change QBFT to also send a prepare when receive a proposal. This also requires changing the quorum needed to be same as the standard quorum used for other QBFT messages.

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java [hunk 1]
```diff
@@ -68,11 +68,11 @@ public void messagesForFutureRoundAreNotActionedUntilRoundIsActive() {
     final ConsensusRoundIdentifier subsequentRoundId = new ConsensusRoundIdentifier(1, 6);
     final RoundSpecificPeers subsequentRoles = context.roundSpecificPeers(subsequentRoundId);
 
-    // required remotely received Prepares = quorum-2
+    // required remotely received Prepares = quorum-1
     // required remote received commits = quorum-1
 
     // Inject 1 too few Commit messages (but sufficient Prepare)
-    for (int i = 0; i < quorum - 3; i++) {
+    for (int i = 0; i < quorum - 2; i++) {
```
Comment (rain-on) [inline consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java:75]: the comment above this line is incorrect now, so should be removed.

File: consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java [hunk 2]
```diff
@@ -68,11 +68,11 @@ public void messagesForFutureRoundAreNotActionedUntilRoundIsActive() {
     final ConsensusRoundIdentifier subsequentRoundId = new ConsensusRoundIdentifier(1, 6);
     final RoundSpecificPeers subsequentRoles = context.roundSpecificPeers(subsequentRoundId);
 
-    // required remotely received Prepares = quorum-2
+    // required remotely received Prepares = quorum-1
```
Comment (rain-on) [inline consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java:71]: can we just ditch these comments all together? Given the quorums are the same

File: consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/LocalNodeNotProposerTest.java [hunk 3]
```diff
@@ -92,6 +95,30 @@ public void preparesReceivedFromNonProposerIsValid() {
     assertThat(context.getCurrentChainHeight()).isEqualTo(1);
   }
 
+  @Test
+  public void prepareFromProposerIsIgnored() {
```
Comment (rain-on) [inline consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/LocalNodeNotProposerTest.java:99]: should this test be removed? I.e. we ACCEPT (not ignore) the prepare from the proposer now.

File: consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java [hunk 4]
```diff
@@ -142,23 +142,28 @@ private void updateStateWithProposalAndTransmit(
...
-        transmitter.multicastPrepare(
-            localPrepareMessage.getRoundIdentifier(), localPrepareMessage.getDigest());
-      } catch (final SecurityModuleException e) {
-        LOG.warn("Failed to create a signed Prepare; {}", e.getMessage());
-      }
+      sendPrepare(block);
+    }
+  }
+
+  private void sendPrepare(final Block block) {
```
Comment (rain-on) [inline consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java:157]: sure we've got tests for this - but confirming, COULD this push us over  "prepared/committed" boundaries (in the case of 1 or 2 validator systems)? And if so - do we appear to operate sensibly?

Comment (jframe) [reply consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftRound.java:157]: Yes we have test for this in QbftRoundTest singleValidatorImportBlocksImmediatelyOnProposalCreation that tests this

File: consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/LocalNodeNotProposerTest.java (hunk 3 above)
Comment (jframe) [reply consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/LocalNodeNotProposerTest.java:99]: Yes it should be. It doesn't make sense now.

File: consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java (hunk 2 above)
Comment (jframe) [reply consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java:71]: done

File: consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java (hunk 1 above)
Comment (jframe) [reply consensus/qbft/src/integration-test/java/org/hyperledger/besu/consensus/qbft/test/FutureRoundTest.java:75]: done

**Review by rain-on — APPROVED** [review]

**Pull request merged**